routes/my_api.js

```diff
@@ -85,6 +86,23 @@ router.get('/parse', user.requireScope('user-read'), iv.validateGET({ q: '?strin
     }).catch(next);
 });
 
+router.post('/converse', user.requireScope('user-exec-command'), (req, res, next) => {
+    let command = req.body.command;
+    if (!command) {
+        next(new BadRequestError('Missing command'));
+        return;
+    }
+
+    Q.try(() => {
```

Comment:
One thing we might consider is replacing the Q library in the future (when we have more time) with native promises. Author seems to have stopped maintaining it a few years ago: https://github.com/kriskowal/q

Reply:
#150